src/_premake_init.lua

```diff
@@ -1577,6 +1589,13 @@
 		staticruntime "Default"
 	end)
 
+	-- 08 April 2018
+
+	api.deprecateValue("flags", "NoFramePointer", 'Use `framepointer "Off"` instead.',
+	function(value)
+			omitframepointer("On")
+	end)
```

Comment:
I believe there's two functions that can be defined, one for `flags { "NoFramePointer" }` and one for `removeflags { "NoFramePointer" }`. See above deprecation for `StaticRuntime`.

Reply:
Well, I commented too soon.
I suppose this is the reason for a Default value?

Reply:
There's this, but sometimes you just want to unset the value. Either it's causing you compile issues (compiler bugs, etc), or you're using a compiler that doesn't fully support the options. I've encountered the latter when people have assumed that Clang accepts all GCC flags, but it actually doesn't.

Reply:
I see.
The problem that I see is that an API like -fvisibility has a Default value that does something (ie `-fvisibility=default`) so I feel that premake should take care of this at a higher level than per-API. Maybe something like the `removeflags` API? Or allowing user to set `nil`?

Reply:
You raise a good point, @starkos @tvandijck @TurkeyMan thoughts on extending the API system to "unset" an API? I like the idea of using the `remove` prefix, it's consistent with the list-based APIs.

Reply:
I prefer a "Default" or "Off" to "remove", since it makes the behavior of that particular switch more explicit. Do you actually need to emit `-fvisibility=default`? Isn't that the same as not including the flag at all?

Reply:
Well I can't find anything about a difference between `-fvisibility=default` and not including the flag so I guess I could no output anything for `visibility "Default"`.

I still stand against using "Off" as an equivalent to remove. In the UnsignedChar API I added, unsignedchar "Off" output `-fno-unsigned-char` and it is necessary as some platforms have unsigned char by default.

As for "Default", it does not sit well with me but I can't find a case that would cause an issue.

Reply:
Also if "Off" is not used as a "Default", it means that boolean APIs must either be changed to string or that the internal workings of boolean APIs accept "Default" as input.

Honestly, I'd rather that all APIs accept "nil" that would simplify things a lot.

Should I open an issue to discuss this? It concerns premake as a whole and not just this PR.